resources/nspawn.go

```diff
@@ -0,0 +1,306 @@
+	machined "github.com/joejulian/go-systemd/machine1"
+)
+
+const (
+	running           = "running"
+	stopped           = "stopped"
+	dbusInterface     = "org.freedesktop.machine1.Manager"
+	machineNew        = "org.freedesktop.machine1.Manager.machineNew"
+	machineRemoved    = "org.freedesktop.machine1.Manager.machineRemoved"
+	nspawnServiceTmpl = "systemd-nspawn@%s"
+)
+
+func init() {
+	gob.Register(&NspawnRes{})
+}
+
+// NspawnRes is an nspawn container resource
+type NspawnRes struct {
+	BaseRes `yaml:",inline"`
+	State   string `yaml:"state"`
+	svc     SvcRes
+}
+
+// Init runs some startup code for this resource
+func (obj *NspawnRes) Init() error {
+	if err := obj.svc.Init(); err != nil {
+		return err
+	}
+	obj.BaseRes.kind = "Nspawn"
+	return obj.BaseRes.Init()
+}
+
+// NewNspawnRes is the constructor for this resource
+func NewNspawnRes(name string, state string) (*NspawnRes, error) {
+	obj := &NspawnRes{
+		BaseRes: BaseRes{
+			Name: name,
+		},
+		State: state,
+	}
+	return obj, obj.Init()
+}
+
+// Validate params
+func (obj *NspawnRes) Validate() error {
+	switch obj.State {
+	case running, stopped:
+		break
+	default:
+		return fmt.Errorf("Invalid State: %s", obj.State)
+	}
+	return nil
+}
+
+// Watch for state changes and sends a message to the bus if there is a change
+func (obj *NspawnRes) Watch(processChan chan event.Event) error {
+	if obj.IsWatching() {
+		return nil
+	}
+	obj.SetWatching(true)
+	defer obj.SetWatching(false)
+	cuid := obj.converger.Register()
+	defer cuid.Unregister()
+
+	var startup bool
+	Startup := func(block bool) <-chan time.Time {
+		if block {
+			return nil // blocks forever
+		}
+		// 1/2 the resolution of converged timeout
+		return time.After(time.Duration(500) * time.Millisecond)
+	}
+
+	// this resource depends on systemd ensure that it's running
+	if !systemdUtil.IsRunningSystemd() {
+		return fmt.Errorf("Systemd is not running.")
+	}
+
+	// create a private message bus
+	bus, err := util.SystemBusPrivateUsable()
+	if err != nil {
+		return fmt.Errorf("Failed to connect to bus: %s", err)
+	}
+
+	// add a match rule to match messages going through the message bus
+	bus.BusObject().Call("org.freedesktop.DBus.AddMatch", 0, fmt.Sprintf(
+		"type='signal',interface='%s',eavesdrop='true'", dbusInterface))
+	buschan := make(chan *dbus.Signal, 10)
+	bus.Signal(buschan)
+
+	var send = false
+	var exit = false
+
+	for {
+		obj.SetState(ResStateWatching)
+		select {
+		// if this resource has been idle for long enough, set converged
+		// to allow the main loop to act on a converged status
+		case <-cuid.ConvergedTimer():
+			cuid.SetConverged(true) // converged!
+			continue
+
+		case event := <-obj.Events():
+			cuid.SetConverged(false)
+			// we avoid sending events on unpause
+			if exit, send = obj.ReadEvent(&event); exit {
+				return nil // exit
+			}
+
+		case <-Startup(startup):
+			cuid.SetConverged(false)
+			send = true
+
+		// process org.freedesktop.machine1 events for this resource's
```

Comment:
the 80 char limit is a suggestion, but if you can make it looks balanced (eg: have the text on one line, have on the other) than it's better keeping it as a hard requirement. I'd balance or have it go long. Sorry nit, I probably didn't explain this well on the first review.